### PR TITLE
Remove deprecations: tock

### PR DESCRIPTION
**Pull request by caguero**
# 🦟 Bug fix

Perform the tock of the tick-tock cycle and remove / move deprecated code.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: Migration.md [hunk 1]
```diff
@@ -5,6 +5,20 @@ Deprecated code produces compile-time warnings. These warning serve as
...
 
+## Gazebo Rendering 9.x to 10.x
+
+### Removals
+
+1. The environment variable `GZ_RENDERING_RESOURCE_PATH` is removed. Use
+`gz::rendering::getResourcePath()` instead.
+
+1. The environment variable `GZ_RENDERING_ENGINE_INSTALL_DIR` is removed. Use
+`gz::rendering::getEngineInstallDir()` instead.
```
Comment (iche033) [inline Migration.md:16]: We're still reading the env vars so just clarifying that it's only the macros that are removed:

```suggestion
1. The macro `GZ_RENDERING_RESOURCE_PATH` is removed. Use
`gz::rendering::getResourcePath()` instead.

1. The macro `GZ_RENDERING_ENGINE_INSTALL_DIR` is removed. Use
`gz::rendering::getEngineInstallDir()` instead.
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -151,8 +151,6 @@ endif()
 set(GZ_RENDERING_ENGINE_RELATIVE_INSTALL_DIR
   ${GZ_LIB_INSTALL_DIR}/gz-${GZ_DESIGNATION}-${PROJECT_VERSION_MAJOR}/engine-plugins
 )
-set(GZ_RENDERING_ENGINE_INSTALL_DIR
-  ${CMAKE_INSTALL_PREFIX}/${GZ_RENDERING_ENGINE_RELATIVE_INSTALL_DIR})
```
Comment (iche033) [inline CMakeLists.txt:155]: I think we may still need to keep this cmake variable here as I see that these are being used in [ogre/src/CMakeLists.txt](https://github.com/gazebosim/gz-rendering/blob/33626b5c48c20d147c167e64fa117043aceec5a5/ogre/src/CMakeLists.txt#L65-L66) and [ogre2/src/CMakeLists.txt](https://github.com/gazebosim/gz-rendering/blob/33626b5c48c20d147c167e64fa117043aceec5a5/ogre2/src/CMakeLists.txt#L90-L91)

Comment (caguero) [reply CMakeLists.txt:155]: Can we replace these calls with `getResourcePath()`?

Comment (iche033) [reply CMakeLists.txt:155]: `getResourcePath()` is a C++ function so we won't be able to replace those calls in CMakeLists.txt. I think we can just keep this cmake var here. Other changes in this PR are fine.

Comment (caguero) [reply CMakeLists.txt:155]: Sorry I meant `GZ_RENDERING_RESOURCE_PATH`, that it's used in some `.cc` files (and in `optix/src/CMakeLists.txt`).

Comment (iche033) [reply CMakeLists.txt:155]: to clarify, I think we are still keeping the environment variable and only deprecating the macro? Or are we deprecating / removing both?

Comment (caguero) [reply CMakeLists.txt:155]: Sorry for the confusion. I think what we want is to remove just the `GZ_RENDERING_RESOURCE_PATH` macro and nothing else. We'll keep supporting the environment variables. [6bd5db5](https://github.com/gazebosim/gz-rendering/pull/1091/commits/6bd5db53959eb69536437e44a1a3fcb8049662d9)

File: Migration.md [hunk 3]
```diff
@@ -12,9 +12,6 @@ release will remove the deprecated code.
 1. The macro `GZ_RENDERING_RESOURCE_PATH` is removed. Use
 `gz::rendering::getResourcePath()` instead.
 
-1. The macro `GZ_RENDERING_ENGINE_INSTALL_DIR` is removed. Use
-`gz::rendering::getEngineInstallDir()` instead.
```
Comment (iche033) [inline Migration.md]: So a little confusing but there are 3 forms of `GZ_RENDERING_ENGINE_INSTALL_DIR`
* cmake variable in CMakeLists.txt
* C++ macro in config.hh.in
* environment variable

We removed the C++ macro (and kept the other two). So we can keep this entry in Migration.md

Comment (caguero) [reply Migration.md]: That's right, thanks. [c383af1](https://github.com/gazebosim/gz-rendering/pull/1091/commits/c383af1a7eed3a95b03613c7221d3358688f9858)

**Review by iche033 — APPROVED** [review]

**Pull request merged**